google/cloud/sqlalchemy_spanner/sqlalchemy_spanner.py

```diff
@@ -54,6 +68,20 @@ class SpannerSQLCompiler(SQLCompiler):
 
     compound_keywords = _compound_keywords
 
+    def visit_empty_set_expr(self, type_):
+        """Return an empty set expression of the given type.
+
+        Args:
+            type_ (sqlalchemy.sql.sqltypes.SchemaType):
+                A SQLAlchemy data type.
+
+        Returns:
+            str: A query to select an empty set of the given type.
+        """
+        return "SELECT CAST(1 AS {}) FROM (SELECT 1) WHERE 1 != 1".format(
+            _type_map_inv[type(type_[0])]
```

Comment:
I'm not familiar with the `SchemaType` but why isn't it (type(type_)) or something similar?

Reply:
In fact I can't say anything straight on it. The method is not documented (even in [source code](https://github.com/sqlalchemy/sqlalchemy/blob/1edbe841bcb43d761f5084a2e77e8f37ee041c35/lib/sqlalchemy/sql/compiler.py#L1894)). The argument is a list with an empty value of some type in it. Few dialects are implementing this method, and they don't try to look at the given type, they just return an empty set of `int` type. It's not correct - I've tried to run several queries, mixingthe types, to ensure it gives errors of a kind `can't cast type1 to int64`.

Thus, I'm taking the first (and the only) element of the given list, take a type of it and then see for a correct type name.

Reply:
I see. I don't like this line from a readability perspective but it seems to work and since the `sqlalchemy.sql.sqltypes.SchemaType` seems to be subscriptable I guess there's no problem